atomic_reactor/plugins/post_rpmqa.py

```diff
@@ -45,17 +45,19 @@ def run(self) -> None:
         self.workflow.data.image_components = self.workflow.build_dir.for_each_platform(
             self.gather_output)
 
-    def gather_output(self, build_dir: BuildDir) -> Optional[List[RpmComponent]]:
+    def gather_output(self, build_dir: BuildDir) -> List[RpmComponent]:
         image = self.workflow.data.tag_conf.get_unique_images_with_platform(build_dir.platform)[0]
         with tempfile.TemporaryDirectory(dir=build_dir.path) as rpmdb_dir:
-            self.workflow.imageutil.extract_file_from_image(image, RPMDB_PATH, rpmdb_dir)
-            rpmdb_path = os.path.join(rpmdb_dir, RPMDB_DIR_NAME)
-
-            if not os.listdir(rpmdb_path):
-                self.log.info('rpmdb directory %s is empty', RPMDB_PATH)
+            try:
+                self.workflow.imageutil.extract_file_from_image(image, RPMDB_PATH, rpmdb_dir)
+            except ValueError as e:
                 if self.workflow.data.dockerfile_images.base_from_scratch:
-                    return None
-                raise RuntimeError(f'rpmdb directory {RPMDB_PATH} is empty')
+                    self.log.info("scratch image doesn't contain or has empty rpmdb %s", RPMDB_PATH)
+                    return []
+                raise RuntimeError(f"rpmdb directory {RPMDB_PATH} in the image, is empty or "
+                                   "doesn't exist") from e
```

Comment:
It seems a bit fragile to assume that "extract_file_from_image() raises ValueError" means "no rpmdb in the image". IMO the extract method either needs a specific exception type for "nothing was extracted" or a param that makes it skip the "nothing was extracted" check

Reply:
refined the handling

Reply:
Nice! :tada: